src/moonlink_datafusion/src/connection_pool.rs

```diff
@@ -0,0 +1,361 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        LazyLock,
+    },
+    time::{Duration, Instant},
+};
+use tokio::{net::UnixStream, sync::Mutex, time::sleep};
+
+struct PooledEntry {
+    stream: UnixStream,
+    last_used: Instant,
+}
+
+static POOL: LazyLock<Mutex<HashMap<String, Mutex<Vec<PooledEntry>>>>> =
```

Comment:
we already have a global lock for `POOL`, curious what's the purpose for mutex inside of each entry?